weasyl/profile.py

```diff
@@ -825,30 +825,33 @@ def do_manage(my_userid, userid, username=None, full_name=None, catchphrase=None
         updates.append('- Catchphrase: %s' % (catchphrase,))
 
     # Birthday
-    if birthday is not None and d.convert_inputdate(birthday):
-        unixtime = d.convert_inputdate(birthday)
-        age = d.convert_age(unixtime)
-
-        d.execute("UPDATE userinfo SET birthday = %i WHERE userid = %i", [unixtime, userid])
-
-        if age < ratings.EXPLICIT.minimum_age:
-            max_rating = ratings.GENERAL.code
-            rating_flag = ""
-        else:
-            max_rating = ratings.EXPLICIT.code
-
-        if d.get_rating(userid) > max_rating:
-            d.engine.execute(
-                """
-                UPDATE profile
-                SET config = REGEXP_REPLACE(config, '[ap]', '', 'g') || %(rating_flag)s
-                WHERE userid = %(user)s
-                """,
-                rating_flag=rating_flag,
-                user=userid,
-            )
-            d._get_all_config.invalidate(userid)
-        updates.append('- Birthday: %s' % (birthday,))
+    if birthday is not None:
+        # HTML5 date format is yyyy-mm-dd
+        split = birthday.split("-")
+        if len(split) == 3 and d.convert_unixdate(day=split[2], month=split[1], year=split[0]):
```

Comment:
Since the admin form requires an explicit “update” checkbox to update something, I think we should fix the behaviour of ignoring invalid inputs and doing nothing. So:

```python
if len(split) != 3 or d.convert_unixdate(day=split[2], month=split[1], year=split[0]) is None:
    raise WeasylError("birthdayInvalid")
```